pil-analyzer/src/statement_processor.rs

```diff
@@ -647,6 +614,86 @@ where
         }
     }
 
+    fn process_struct_declaration(
+        &mut self,
+        symbol: Symbol,
+        struct_decl: StructDeclaration<parsed::Expression>,
+    ) -> Vec<PILItem> {
+        let type_vars = struct_decl.type_vars.vars().collect();
+        let fields = struct_decl
+            .fields
+            .into_iter()
+            .map(|v| NamedType {
+                name: v.name.clone(),
```

Comment:
Why do you need to clone here? Maybe you need a destructuring assignment at the start of the function
```
let StructDeclaration{name, type_vars, fields} = struct_decl;
```